src/main/java/org/mitre/synthea/modules/covid/C19ImmunizationModule.java

```diff
@@ -0,0 +1,312 @@
+  // exponentially.
+  public static final String C19_LATE_ADOPTER_MODEL = "C19_LATE_ADOPTER_MODEL";
+
+  // This is somewhat redundant given that there is C19_VACCINE_STATUS, but GMF modules can't
+  // check attributes set to java enumeration values, so this will just be a simple boolean
+  public static final String C19_FULLY_VACCINATED = "C19_FULLY_VACCINATED";
+
+  public enum VaccinationStatus {
+    NOT_ELIGIBLE,
+    WAITING_FOR_SHOT,
+    POTENTIAL_LATE_ADOPTER,
+    NEVER_GOING_TO_GET_SHOT,
+    FIRST_SHOT,
+    FULLY_VACCINATED
+  }
+
+  /**
+   * Create and initialize am instance of the module.
+   */
+  public C19ImmunizationModule() {
+    C19VaccineAgeDistributions.initialize();
+    C19Vaccine.initialize();
+    this.name = "COVID-19 Immunization Module";
+  }
+
+  public Module clone() {
+    return this;
+  }
+
+  /**
+   * Checks a person's health record to see if they currently have COVID-19.
+   * @param person the person to check
+   * @return true if there is an active COVID-19 condition on the record.
+   */
+  public static boolean currentlyHasCOVID(Person person) {
+    return person.record.conditionActive(COVID_CODE);
+  }
+
+  /**
+   * Check to see if a person is eligible for a COVID-19 vaccine at the time in the simulation.
+   * The check is based on date (when a vaccine was given an EUA) and age
+   * @param person the person to check
+   * @param time current time in the simulation
+   * @return true if the person would be eligible for a vaccine at the point in the simulation
+   */
+  public static boolean eligibleForShot(Person person, long time) {
+    int age = person.ageInYears(time);
+    if (age >= FIRST_ELIGIBLE_AGE
+        || (time >= EXPAND_AGE_TO_TWELVE) && age >= EXPANDED_ELIGIBLE_AGE) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Given an individual, randomly select whether they will get vaccinated, with the selection
+   * weighted by the proportion of individuals in their age group who have been vaccinated according
+   * to national statistics.
+   * @param person the person to check
+   * @param time current time in the simulation
+   * @return true if the person should proceed to vaccination
+   */
+  public static boolean decideOnShot(Person person, long time) {
+    double chanceOfGettingShot = C19VaccineAgeDistributions.chanceOfGettingShot(person, time);
+    return chanceOfGettingShot >= person.rand();
+  }
+
+  /**
+   * Select a vaccine for an individual based on national distribution statistics and whether the
+   * vaccine was eligible for the time in simulation.
+   * @param person the person to check
+   * @param time current time in the simulation
+   * @return a vaccine that should be administered to the person
+   */
+  public static C19Vaccine.EUASet selectVaccine(Person person, long time) {
+    if (time < MODERNA_APPROVED) {
+      return C19Vaccine.EUASet.PFIZER;
```

Comment:
Strangely, this method will return the Pfizer vaccine even if the `time` is January 1777 and George Washington's army can select the Pfizer vaccine while they are simultaneously being vaccinated for small pox.

https://www.loc.gov/rr/scitech/GW&smallpoxinoculation.html

Reply:
And yes, I know the `process` method makes this impossible. But this is a `public static` method.

Reply:
Will now throw an `IllegalArgumentException` if provided with a time before the Pfizer vaccine was approved. Apologies to the Continental Army.